### PR TITLE
highgui: wayland: show "NO" status if dependency is missing

### LINKED ISSUE #25495: highgui: wayland: if dependency is missing, "NO" message will helps to install them.
### Describe the feature and motivation

- OpenCV 4.x branch ( https://github.com/opencv/opencv/commit/2cd330486ec4597eab49c1575fc4a6603f205a6a )
- Ubuntu 24.04
- GCC 13.2

Evenif dependency is missing, we cannnot know what dependency is missing.
 
https://github.com/opencv/opencv/blob/2cd330486ec4597eab49c1575fc4a6603f205a6a/CMakeLists.txt#L1368-L1384

It is better to show "NO" if dependency libraries are missing.

```diff
- if(HAVE_WAYLAND_CLIENT)
-   status("    Wayland Client:" "YES (ver ${WAYLAND_CLIENT_VERSION})")
- endif()
+ status("    Wayland Client:" HAVE_WAYLAND_CLIENT THEN "YES (ver ${WAYLAND_CLIENT_VERSION})" ELSE "NO")
```

Case) WITH_WAYLAND is not set.
```plaintext
  GUI:                           NO
    GTK+:                        NO
    VTK support:                 NO
```

Case) `WITH_WAYLAND` is set, and all dependencies are installed.
```plaintext
  GUI:                           Wayland
    Wayland:                     (Experimental) YES
      Wayland Client:            YES (ver 1.22.0)
      Wayland Cursor:            YES (ver 1.22.0)
      Wayland Protocols:         YES (ver 1.34)
      Xkbcommon:                 YES (ver 1.6.0)
      Wayland EGL(Option):       NO
    GTK+:                        NO
    VTK support:                 NO
```

Case) `WITH_WAYLAND` is set, but `Xkbcommon` is missing, so Wayland backend is disabled.
```plaintext
  GUI:                           NO
    Wayland:                     NO
      Wayland Client:            YES (ver 1.22.0)
      Wayland Cursor:            YES (ver 1.22.0)
      Wayland Protocols:         YES (ver 1.34)
      Xkbcommon:                 NO
      Wayland EGL(Option):       NO
    GTK+:                        NO
    VTK support:                 NO
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kumataro**
Close #25495 

- [doc] Add document to enable Wayland highgui-backend in ubuntu 24.04.
- [build] Show "NO" status instead of version if dependency library is missing.
- [build] Fix to find Wayland EGL.
- [fix] Add some callback stub functions to suppress build warning.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: doc/tutorials/app/highgui_wayland_ubuntu.markdown [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+Setup
+-----
+- Setup Ubuntu 24.04.
+- `sudo apt install build-essential git cmake` to build OpenCV.
+- `sudo apt install libwayland-dev wayland-protocol libxkbcommon-dev` to enable Wayland highgui-backend.
+- (Option) `sudo apt install ninja-build` (or remove `-GNinja` option for cmake command).
+- (Option) `sudo apt install libwayland-egl1` to enable Wayland EGL library.
+
+Get OpenCV from github
```
[the remark this answers is not in the document]
Comment (Kumataro) [reply doc/tutorials/app/highgui_wayland_ubuntu.markdown]: Thank you for your review! I fixed it.

File: doc/tutorials/app/highgui_wayland_ubuntu.markdown [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+
+```bash
+// g++ main.cpp -o a.out -I /usr/local/opencv4 -lopencv_core -lopencv_highghui
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+
+int main(void)
+{
+  cv::Mat src(240, 320, CV_8UC3, cv::Scalar::all(128) );
```
[the remark this answers is not in the document]
Comment (Kumataro) [reply doc/tutorials/app/highgui_wayland_ubuntu.markdown]: Yes, I agree that gray image is not suitable to test, Thank you!

And there are some negative opinions about using lena image. Therefore, we use opencv-logo.png as an alternative.

File: doc/tutorials/app/highgui_wayland_ubuntu.markdown [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      cv::imshow("src", src );
+      key = cv::waitKey(50);
+  } while( key != 'q' );
+  return 0;
+}
+```
+
+Limitation/Known problem
+------------------------
+- If calling `namedWindow()` is missing, `imshow()` doesn't work (maybe bug).
```
[the remark this answers is not in the document]
Comment (Kumataro) [reply doc/tutorials/app/highgui_wayland_ubuntu.markdown]: I create https://github.com/opencv/opencv/issues/25497 .

I test with GTK3 backend, it works well. 
Maybe only Wayland backend, it happens.

File: doc/tutorials/app/highgui_wayland_ubuntu.markdown (hunk 1 above)
[the remark this answers is not in the document]
Comment (opencv-alalek) [reply doc/tutorials/app/highgui_wayland_ubuntu.markdown]: `GitHub`

https://en.wikipedia.org/wiki/GitHub

[the remark this answers is not in the document]
Comment (Kumataro) [reply doc/tutorials/app/highgui_wayland_ubuntu.markdown]: Thank you for your point,, I fixed it.

File: doc/tutorials/app/highgui_wayland_ubuntu.markdown [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+- `sudo apt install build-essential git cmake` to build OpenCV.
+- `sudo apt install libwayland-dev wayland-protocol libxkbcommon-dev` to enable Wayland highgui-backend.
+- (Option) `sudo apt install ninja-build` (or remove `-GNinja` option for cmake command).
+- (Option) `sudo apt install libwayland-egl1` to enable Wayland EGL library.
+
+Get OpenCV from Github
+----------------------
+
+```bash
+cd work
```
[the remark this answers is not in the document]
Comment (Kumataro) [reply doc/tutorials/app/highgui_wayland_ubuntu.markdown]: Yes, I fixed it, thank you !

**Pull request merged**